lib/util/package-rules/files.ts

```diff
@@ -1,23 +1,27 @@
 import is from '@sindresorhus/is';
+import { minimatch } from 'minimatch';
 import type { PackageRule, PackageRuleInputConfig } from '../../config/types';
 import { Matcher } from './base';
 
-export class FilesMatcher extends Matcher {
+export class FileNamesMatcher extends Matcher {
   override matches(
     { packageFile, lockFiles }: PackageRuleInputConfig,
-    { matchFiles }: PackageRule
+    { matchFileNames }: PackageRule
   ): boolean | null {
-    if (is.undefined(matchFiles)) {
+    if (is.undefined(matchFileNames)) {
       return null;
     }
     if (is.undefined(packageFile)) {
       return false;
     }
 
-    return matchFiles.some(
-      (fileName) =>
-        packageFile === fileName ||
-        (is.array(lockFiles) && lockFiles?.includes(fileName))
+    return matchFileNames.some(
+      (matchFileName) =>
+        minimatch(packageFile, matchFileName, { dot: true }) ||
```

Comment:
should we create a Minimatch object outside the loop, so Minimatch doesn't need to parse and create the regex on every iteration?

Reply:
Not worth it because it should always be <=3 total files (packageFile + lockFile)